mvpa2/measures/anova.py

```diff
@@ -79,12 +79,12 @@ def _call(self, dataset):
         ul = targets_sa.unique
 
         na = len(ul)
-        bign = float(dataset.nsamples)
         alldata = dataset.samples
 
         # total squares of sums
         sostot = np.sum(alldata, axis=0)
         sostot *= sostot
+        bign = sostot.dtype.type(dataset.nsamples)
```

Comment:
On Mon, 11 Jan 2016, Michael Hanke wrote:
>  >          alldata = dataset.samples

>  >          # total squares of sums
>  >          sostot = np.sum(alldata, axis=0)
>  >          sostot *= sostot
>  > +        bign = sostot.dtype.type(dataset.nsamples)

>    Ah, right. Didn't think about that. So we are left with a full copy on
>    div...

only if type is not float ;)